### DISCUSSION TITLE
IFrame embed model viewer 3D works AR not work because of Content Security Policy(Android)

### DISCUSSION THREAD

**Discussion by base-ddf** [google/model-viewer#3283] [Q&A]
### Description

When trying to open a 3d page as an iframe on another site as an embed code, a csp error is encountered when trying to open it as AR. **I want to ask if it can be solved at client-side?**Don't want to pointless effort if it can not.

Including Scene-viewer intent script for AR, Content Policy blocks something.  I cant figure out. Tried CSP headers and scripts for that, none of them worked.

Android side only. Usdz intent is working well suprisely.

I would be glad if you help

#### Live Demo
[Embeded site](https://madamecoco.basebros.com/8682116226738.html)

[the one use iframe](https://www.madamecoco.com/raina-sacakli-dokuma-kilim-lacivert-60x100cm-yg-343-783/)

![image](https://user-images.githubusercontent.com/91538448/157540562-440ac4db-5aaa-4104-8338-ccfdcff704bb.png)

3D work referance image
![image](https://user-images.githubusercontent.com/91538448/157541622-9c3e339f-1cf7-4396-b976-82adb3ba5c40.png)

AR not work referance image
![image](https://user-images.githubusercontent.com/91538448/157541839-03fbd87f-b6bb-446c-9cfc-5a63d9067420.png)

### AR

- [x ] WebXR
- [x ] SceneViewer
- [x ] QuickLook

**Comment by samaneh-kazemi** [comment]
Hi @base-ddf, thanks for reaching out. You need to specifically grant the iframe XR permissions with `xr-spatial-tracking`: https://modelviewer.dev/examples/augmentedreality/#ar

Look at these similar discussions: #3209 & #3057
